pkg/activator/throttler_test.go

```diff
@@ -0,0 +1,209 @@
+	added := true
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+
+	response := throttler.AddIfAbsent(revID)
+	got := len(throttler.breakers)
+	if got != want {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", want, got)
+	}
+	if added != response {
+		t.Errorf("Unexpected response indicating the existance of the breaker. Want %t, got %t", added, response)
+	}
+}
+
+func TestThrottler_AddAlreadyExists(t *testing.T) {
+	want := 1
+	added := false
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+
+	throttler.AddIfAbsent(revID)
+	response := throttler.AddIfAbsent(revID)
+	got := len(throttler.breakers)
+	if got != want {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", want, got)
+	}
+	if added != response {
+		t.Errorf("Unexpected response indicating the existance of the breaker. Want %t, got %t", added, response)
+	}
+}
+
+func TestThrottler_UpdateCapacity(t *testing.T) {
+	want := int32(10)
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+	throttler.AddIfAbsent(revID)
+	throttler.UpdateCapacity(revID, 1)
+	breaker, _ := throttler.breakers[revID]
+	got := breaker.Capacity()
+	if got != want {
+		t.Errorf("Unexpected capacity of the breaker. Want %d, got %d", want, got)
+	}
+}
+
+func TestThrottler_UpdateCapacity_NonExistingRevision(t *testing.T) {
+	throttler := getThrottler(false, defaultMaxConcurrency, t)
+	throttler.AddIfAbsent(revID)
+	err := throttler.UpdateCapacity(revID, 1)
+	if err.Error() != fmt.Sprintf("no revision was found for: %s", revID) {
+		t.Errorf("Unexpected error occured. Got %s", err.Error())
+	}
+}
+
+func TestThrottler_UpdateCapacity_NonExistingService(t *testing.T) {
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+	err := throttler.UpdateCapacity(revID, 1)
+	if err.Error() != fmt.Sprintf("no service name registered for the revision: %s", revID) {
+		t.Errorf("Unexpected error occured. Got %s", err.Error())
+	}
+}
+
+func TestThrottler_Remove(t *testing.T) {
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+	throttler.AddIfAbsent(revID)
+	got := len(throttler.breakers)
+	if got != 1 {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", 1, got)
+	}
+	throttler.Remove(revID)
+	got = len(throttler.breakers)
+	if got != 0 {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", 0, got)
+	}
+}
+
+func TestThrottler_Exists(t *testing.T) {
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+	throttler.AddIfAbsent(revID)
+	exists := throttler.Exists(revID)
+	if !exists {
+		t.Errorf("Revision doesn't exist.")
+	}
+}
+
+func TestThrottler_Try(t *testing.T) {
+	var got int32
+	want := int32(1)
+	throttler := getThrottler(true, defaultMaxConcurrency, t)
+	throttler.AddIfAbsent(revID)
+	throttler.UpdateCapacity(revID, 1)
+	throttler.Try(revID, func() {
+		got++
+	})
+	if got != want {
+		t.Errorf("Unexpected number of function runs in Try. Want %d, got %d", want, got)
+	}
+}
```

Comment:
Can we add a test that `Try`s for different revisions and asserts that even when one of them is already full, the other can still work?